File: 1-js/1-js-basics/3.2-conditional-statements-advanced-exercise/03-newHouse.js
function solve([type, num, budget]) {

    num = Number(num);

    budget = Number(budget);

    let data = {

        'Roses': 5,

        'Dahlias': 3.80,

        'Tulips': 2.80,

        'Narcissus': 3,

        'Gladiolus': 2.50

    }

    let total = num * data[type];

    if (type == 'Roses' && num > 80) {

        total *= 0.90;

    } else if ((type == 'Dahlias' && num > 90) || (type == 'Tulips' && num > 80)) {

        total *= 0.85;

    } else if (type == 'Narcissus' && num < 120) {

        total *= 1.15;

    } else if (type == 'Gladiolus' && num < 80) {

        total *= 1.20;

    }

    return total <= budget ? `Hey, you have a great garden with ${num} ${type} and ${(budget - total).toFixed(2)} leva left.` : `Not enough money, you need ${(total - budget).toFixed(2)} leva more.`;
       
}

console.log(solve(['Narcissus', '119', '360']));

File: 2-python/1-python-basics/3.2-conditional-statements-advanced-exercise/03-new_house.py
flowers_type = input()
flowers_count = int(input())
budget = int(input())
rose_price = 5
dahlia_price = 3.8
tulip_price = 2.8
narcissus_price = 3
gladiolus_price = 2.5
price = 0

if flowers_type == 'Roses':
    price = rose_price * flowers_count
    if flowers_count > 80:
        price *= 0.9
elif flowers_type == 'Dahlias':
    price = dahlia_price * flowers_count
    if flowers_count > 90:
        price *= 0.85
elif flowers_type == 'Tulips':
    price = tulip_price * flowers_count
    if flowers_count > 80:
        price *= 0.85
elif flowers_type == 'Narcissus':
    price = narcissus_price * flowers_count
    if flowers_count < 120:
        price *= 1.15
else:
    price = gladiolus_price * flowers_count
    if flowers_count < 80:
        price *= 1.20

result = budget - price

if result >= 0:
    print(f'Hey, you have a great garden with {flowers_count} {flowers_type} and {result:.2f} leva left.')
else:
    print(f'Not enough money, you need {abs(result):.2f} leva more.')
